plugins/mobile_channel_header_plug/mobile_channel_header_plug.jsx

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export default class MobileChannelHeaderPlug extends React.PureComponent {
```

Comment:
It'd be difficult to abstract them out unfortunately. I could throw them into the same "Plug" component but it would just be moving 90% of this code into the other file.

Reply:
Maybe we can do that, my only concern is removing power from the builders hands. What if I want to add something more than just a small button or in a different style? Maybe I want it to have some text on it or I want the dropdown component to have a lot more information on it

Plugins are about giving power, so that's the reasoning behind this

Reply:
Discussion continued here: https://pre-release.mattermost.com/core/pl/qodeku56x3bombzpwuanwo1qxh